k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapConnectionTest.java

```diff
@@ -75,17 +75,37 @@ public void setUp() throws Exception {
     }
 
     @Test
-    public void open_withCapabilitiesInInitialResponse_shouldNotIssueCapabilitiesCommand() throws Exception {
+    public void open_withNoCapabilitiesInInitialResponse_shouldIssuePreAuthCapabilitiesCommand() throws Exception {
+        settings.setAuthType(AuthType.PLAIN);
+        MockImapServer server = new MockImapServer();
+        server.output("* OK example.org server");
+        server.expect("1 CAPABILITY");
+        server.output("* CAPABILITY IMAP4 IMAP4REV1 AUTH=PLAIN");
+        server.output("1 OK CAPABILITY Completed");
+        server.expect("2 AUTHENTICATE PLAIN");
+        server.output("+");
+        server.expect(ByteString.encodeUtf8("\000" + USERNAME + "\000" + PASSWORD).base64());
+        server.output("2 OK Success");
+        simplePostAuthenticationDialog(server, true, 3);
+
+        ImapConnection imapConnection = startServerAndCreateImapConnection(server);
+
+        imapConnection.open();
+
+        server.verifyConnectionStillOpen();
+        server.verifyInteractionCompleted();
+    }
+
+    @Test
+    public void open_withCapabilitiesInInitialResponse_shouldNotIssuePreAuthCapabilitiesCommand() throws Exception {
         settings.setAuthType(AuthType.PLAIN);
         MockImapServer server = new MockImapServer();
         server.output("* OK [CAPABILITY IMAP4 IMAP4REV1 AUTH=PLAIN]");
         server.expect("1 AUTHENTICATE PLAIN");
         server.output("+");
         server.expect(ByteString.encodeUtf8("\000" + USERNAME + "\000" + PASSWORD).base64());
         server.output("1 OK Success");
-        server.expect("2 LIST \"\" \"\"");
-        server.output("* LIST () \"/\" foo/bar");
-        server.output("2 OK");
+        simplePostAuthenticationDialog(server, true, 2);
```

Comment:
This seems like it just breeds method explosion to work around the lack of named arguments in Java.